tests/0033-regex_subscribe.c

```diff
@@ -304,33 +308,48 @@ static int do_test(const char *assignor) {
         testid = test_id_generate();
         test_str_id_generate(groupid, sizeof(groupid));
 
-        rd_snprintf(topics[0], sizeof(topics[0]), "%s_%s",
-                    test_mk_topic_name("regex_subscribe_TOPIC_0001_UNO", 0),
-                    groupid);
-        rd_snprintf(topics[1], sizeof(topics[1]), "%s_%s",
-                    test_mk_topic_name("regex_subscribe_topic_0002_dup", 0),
-                    groupid);
-        rd_snprintf(topics[2], sizeof(topics[2]), "%s_%s",
-                    test_mk_topic_name("regex_subscribe_TOOTHPIC_0003_3", 0),
-                    groupid);
+        /* Generate unique test run ID for topic isolation to prevent
+         * cross-test contamination from leftover topics */
+        char *test_run_id = rd_strdup(test_str_id_generate_tmp());
+
+        rd_snprintf(
+            topics[0], sizeof(topics[0]), "%s",
+            test_mk_topic_name(
+                tsprintf("regex_subscribe_TOPIC_0001_UNO_%s", test_run_id), 0));
+        rd_snprintf(
+            topics[1], sizeof(topics[1]), "%s",
+            test_mk_topic_name(
+                tsprintf("regex_subscribe_topic_0002_dup_%s", test_run_id), 0));
+        rd_snprintf(
+            topics[2], sizeof(topics[2]), "%s",
+            test_mk_topic_name(
+                tsprintf("regex_subscribe_TOOTHPIC_0003_3_%s", test_run_id),
+                0));
 
         /* To avoid auto topic creation to kick in we use
          * an invalid topic name. */
-        rd_snprintf(
-            nonexist_topic, sizeof(nonexist_topic), "%s_%s",
-            test_mk_topic_name("regex_subscribe_NONEXISTENT_0004_IV#!", 0),
-            groupid);
+        rd_snprintf(nonexist_topic, sizeof(nonexist_topic), "%s",
+                    test_mk_topic_name(
+                        tsprintf("regex_subscribe_NONEXISTENT_0004_IV#!_%s",
+                                 test_run_id),
+                        0));
 
         /* Produce messages to topics to ensure creation. */
-        for (i = 0; i < topic_cnt; i++)
+        for (i = 0; i < topic_cnt; i++) {
+                test_create_topic_if_auto_create_disabled(NULL, topics[i], 1);
+                test_wait_topic_exists(NULL, topics[i], tmout_multip(10000));
```

Comment:
This should be after produce, in the case where we're relying on autocreate. 
But instead of this why not just create the topics explicitly all the time, would that be possible and easier?